firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/CustomModel.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.ml.modeldownloader;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.io.File;
+
+/**
+ * Used to store information about custom models that are being downloaded or are already downloaded
+ * on a device.
+ */
+public class CustomModel {
+  private String name;
+  private long downloadId;
+  private long fileSize;
+  private String modelHash;
+  private String localFilePath = "";
```

Comment:
Is the class intended to be mutated? Why not make all fields `final`?

Reply:
Added a comment to the class. At this point I'm just adding the public parts - enough to get the API headers written in the next PR. The file associated with the model can be updated but will only be supported as a protected method call via the downloader. So no public way to modify the class. The name however should be final. There will be a few other update calls and private fields in the next few PRs.

Reply:
Making this class mutable and updating it as model state changes has many consequences:

1. It must be made thread-safe as updates to it may happen on a thread different from where the developer is using it.
2. You would also have to keep track of all instances of `CustomModel` that are being used and update them all.

So my question is, is it worth it? Can the class be made immutable and for any state updates a new `CustomModel` instance can be created and handed out to progress listeners?